goth/runner/probe/__init__.py

```diff
@@ -324,6 +325,9 @@ async def run_command_on_host(
                 )
                 yield cmd_task, cmd_monitor
 
+                await cmd_task
```

Comment:
It's important to wait until `cmd_task` finishes here, in the `monitored_logger`'s scope. Otherwise we're risking that the monitor gets disconnected from the tasks output (which happens when exiting the `monitored_logger` context manager) before seeing all lines of the output (this happened in the unit test for `run_command_on_host` added in this PR, before adding this change).